hashing/te-tag-query-python/all_in_one_lib/commands/label.py

```diff
@@ -0,0 +1,77 @@
+##!/usr/bin/env python
+
+import typing as t
+
+import TE
+
+from ..collab_config import CollaborationConfig
+from ..content_type import meta, text
+from ..dataset import Dataset
+from . import base
+
+
+class LabelCommand(base.Command):
+    """
+    Apply labels to items in ThreatExchange.
+
+    Examples:
+
+    Label text
+      all-in-one -c te.cfg label violating_label_from_config,other_label text "this is an example bad text"
+
+    Label descriptor
+      all-in-one -c te.cfg label false-positive descriptor 12345
+    """
+
+    FALSE_POSITIVE = "false_positive"
+
+    @classmethod
+    def init_argparse(cls, ap) -> None:
+        ap.add_argument(
+            "labels",
+            type=lambda s: s.strip().split(","),
+            metavar="C  SV",
+            help="labels to apply to item",
+        )
+        ap.add_argument(
+            "content_type", choices=["descriptor"], help="what content type to label"
+        )
+        ap.add_argument("descriptor_id", help="the id of a descriptor to label")
+
+    @classmethod
+    def init_from_namespace(cls, ns) -> "LabelCommand":
+        return cls(ns.descriptor_id, ns.labels)
+
+    def __init__(self, descriptor_id: int, labels: t.List[str]) -> None:
+        self.descriptor_id = descriptor_id
+        self.labels = labels
+        self.false_positive = False
+        # Only use reaction logic if not also adding true positive labels
+        if self.FALSE_POSITIVE in labels:
+            self.labels.remove(self.FALSE_POSITIVE)
+            if not self.labels:
+                self.false_positive = True
+
+    def execute(self, dataset: Dataset) -> None:
+        params = {
+            "descriptor_id": self.descriptor_id,
+            "privacy_type": "HAS_PRIVACY_GROUP",
+            "privacy_members": cfg.privacy_groups,
+        }
+
+        if self.false_positive:
+            params["reactions"] = "DISAGREE_WITH_TAGS"
+        else:
+            params["tags"] = self.labels
+        # TODO: Handle gracefully target doesn't exist
+        # TODO: Handle already labeled (merge don't stomp)
+        err_message, ex, response = TE.Net.copyThreatDescriptor(params, showURLs=True, dryRun=False)
+        if ex:
+            raise ex
+        if err_message:
+            raise base.CommandError(err_message)
+        if not response:
+            raise base.CommandError("Mystery error - empty response")
```

Comment:
This whole thing actually doesn't work at all. Consider just replacing with notimplement error or implement false_positive rxn only as first step.